### ISSUE TITLE
Add option for keeping the video element when using OIPF or HTML5 players

### ISSUE THREAD

**Issue by anttijaatsi** [HbbTV-Association/ReferenceApplication#72]
## Summary

The reference application should have at least a setting that keeps the video player element in the DOM tree after clearing a video that the same element is used for all video plays (if mode not changed). This could be used by developers and manufacturers to test devices if there is a difference how they behave whether the video element is removed after every play or not. 

In my experience there are differences in some devices whether the element is kept or removed. The reference application could be an impartial application for testing this.

 ## Steps to reproduce

1. Set mode to HbbTV 1.5 / HbbTV 2.0.x
2. Start any video
3. Stop the video
4. Start a new video

## What is the current behavior? 

When a video is stopped the player element is removed from the DOM tree by the clearVideo -function. There is no option for keeping the video element.

## What is the expected correct behavior?

There would be a setting in the settings view for example "Keep player element". If this setting is switched on, clearVideo -function won't remove the player element from the DOM tree.

## Relevant data

OIPF Player: https://git.sofiadigital.fi/sofia/ReferenceApplication/-/blob/master/src/videoplayer/videoplayer_oipf.js#L652
HTML5 Player: https://git.sofiadigital.fi/sofia/ReferenceApplication/-/blob/master/src/videoplayer/videoplayer_html5.js#L904

## Possible fixes

`
if (!keepPlayerElement) {
  $( "#video" ).remove(); // clear from dom
}
`

**Comment by jpiesing** [comment]
Is there a reason why everyone would benefit from this? Is there a reason why integrating this into the trunk would be better than you just making your own branch for your own purposes?
Not objecting, just asking.

**Comment by Murmur** [comment]
@anttijaatsi which one works best? Removing a player(oipfAV obj, html video tag) or reusing an instance over the entire period of app lifetime?

Apps hold an object "parked" in DOM even if view navigation is back to a menu/catalog/whatever "normal* web page text content?

**Comment by anttijaatsi** [comment]
> Is there a reason why everyone would benefit from this? Is there a reason why integrating this into the trunk would be better than you just making your own branch for your own purposes?

In my observations there are differences how some of the devices work whether the player element is kept or removed. Manufacturers could test their devices in both cases. Developers could use the reference application to test cases where they notice that not all devices work in their application the same way.

> which one works best? Removing a player(oipfAV obj, html video tag) or reusing an instance over the entire period of app lifetime?

I have noticed that there are devices that need to keep the player element. There are also some devices that probably work better if the player element is removed every time. In my opinion the player element should be created only once and used for the app life time.

> Apps hold an object "parked" in DOM even if view navigation is back to a menu/catalog/whatever "normal* web page text content?

Yes, if the "Keep player element" is switched on, then the player should be created only once if the mode is not switched.

**Comment by juhajoki** [comment]
@anttijaatsi is this still something you'd like to see implemented in trunk?

**Comment by juhajoki** [comment]
closing, curated comments to be added as informative for developer considerations.
